python/dgl/nn/pytorch/conv/grouprevres.py

```diff
@@ -0,0 +1,186 @@
+        outputs = ctx.outputs.pop()
+
+        # reconstruct input node features
+        with torch.no_grad():
+            # inputs[0] used to have the input node features
+            inputs_inverted = ctx.fn_inverse(*((outputs,)+inputs[1:]))
+            # clear memory of outputs
+            outputs.storage().resize_(0)
+
+            x = inputs[0]
+            x.storage().resize_(int(np.prod(x.size())))
+            x.set_(inputs_inverted)
+
+        # compute gradients
+        with torch.set_grad_enabled(True):
+            detached_inputs = []
+            for i, element in enumerate(inputs):
+                if isinstance(element, torch.Tensor):
+                    element = element.detach()
+                    element.requires_grad = ctx.input_requires_grad[i]
+                detached_inputs.append(element)
+
+            detached_inputs = tuple(detached_inputs)
+            temp_output = ctx.fn(*detached_inputs)
+
+        filtered_detached_inputs = tuple(filter(lambda x: x.requires_grad, detached_inputs))
+        gradients = torch.autograd.grad(outputs=(temp_output,),
+                                        inputs=filtered_detached_inputs + ctx.weights,
+                                        grad_outputs=grad_outputs)
+
+        input_gradients = []
+        i = 0
+        for rg in ctx.input_requires_grad:
+            if rg:
+                input_gradients.append(gradients[i])
+                i += 1
+            else:
+                input_gradients.append(None)
+
+        gradients = tuple(input_gradients) + gradients[-len(ctx.weights):]
+
+        return (None, None, None) + gradients
+
+
+class GroupRevRes(nn.Module):
+    r"""Grouped reversible residual connections for GNNs, as introduced in
+    `Training Graph Neural Networks with 1000 Layers <https://arxiv.org/abs/2106.07476>`__
+
+    Parameters
+    ----------
+    gnn_module : nn.Module
+        GNN module for message passing. The input and output node representation size need
+        to be the same.
+    group : int, optional
+        The input node features will be partitioned into the specified number of groups.
+
+    Examples
+    --------
+
+    >>> import dgl
+    >>> import torch
+    >>> from dgl.nn import GraphConv, GroupRevRes
+
+    >>> num_nodes = 5
+    >>> num_edges = 20
+    >>> feats = 32
+    >>> group = 2
+    >>> g = dgl.rand_graph(num_nodes, num_edges)
+    >>> h = torch.randn(num_nodes, feats)
+    >>> conv = GraphConv(feats // group, feats // group)
+    >>> model = GroupRevRes(conv, group)
+    >>> out = model(g, h)
+    """
+    def __init__(self, gnn_module, group=2):
+        super(GroupRevRes, self).__init__()
+        self.gnn_modules = nn.ModuleList()
+        for i in range(group):
+            if i == 0:
+                self.gnn_modules.append(gnn_module)
+            else:
+                self.gnn_modules.append(deepcopy(gnn_module))
+        self.group = group
+
+    def _forward(self, x, g, *args):
```

Comment:
Suggest use `(self, g, x, *args)` instead of `(self, x, g, *args)` for a consistent style. Similar suggestion to `_inverse`

Reply:
I feel having `*args` here is quite confusing and not obvious why we need it. Consider simplify it away.

Reply:
Changed `(self, x, g, *args)` -> `(self, g, x, *args)`.